packages/database/src/core/BaseDatabase.js

```diff
@@ -71,16 +68,17 @@ const RAW_INPUT_PATTERN = /(^CASE)|(^to_timestamp)/;
 // keeping all the tests passing
 const HANDLER_DEBOUNCE_DURATION = 250;
 
-export class TupaiaDatabase {
-  /**
-   * @param {TupaiaDatabase} [transactingConnection]
-   * @param {DatabaseChangeChannel} [transactingChangeChannel]
-   */
-  constructor(transactingConnection, transactingChangeChannel, useNumericStuff = false) {
+export class BaseDatabase {
```

Comment:
Gist 3: Since `pg-pubsub` is now supported in browser/pglite, all the change handlers are not relevant to front end. Hence, I've introduced this parent class `BaseDatabase`, abstract the logic so that that server and browser can extend from:
- `TupaiaDatabase` (server) - extra logic of change handlers
-  `DatatrakDatabase` (browser) - extra logic for PGLite connection